### PR TITLE
fix p2p unit-tests flake #246

### PR DESCRIPTION
**Pull request by adecaro**
Signed-off-by: Angelo De Caro <adc@zurich.ibm.com>

### CODE REVIEW

File: platform/view/services/comm/p2p_test.go [hunk 1]
```diff
@@ -105,7 +113,16 @@ func setupTwoNodesFromFiles(t *testing.T) (*P2PNode, *P2PNode, string, string) {
...
-	bootstrapNode, node := setupTwoNodes(t, bootstrapNodeID, bootstrapNodeEndpoint, nodeID, nodeEndpoint, &PrivateKeyFromFile{bootstrapNodeSK}, &PrivateKeyFromFile{nodeSK})
+	// sometimes setupTwoNodes can fail due to timeouts, give it a second chance
+	var err error
+	var bootstrapNode, node *P2PNode
+	for i := 0; i < 3; i++ {
+		bootstrapNode, node, err = setupTwoNodes(t, bootstrapNodeID, bootstrapNodeEndpoint, nodeID, nodeEndpoint, &PrivateKeyFromFile{bootstrapNodeSK}, &PrivateKeyFromFile{nodeSK})
+		if err == nil {
+			break
+		}
+	}
```
Comment (mbrandenburger) [inline platform/view/services/comm/p2p_test.go:124]: I suggest a dedicated Retry function, which can be used in many places.

This is cleaner than just running the code in the loop adding a comment that we want to retry ... 

For instance:
```go
func retry(attempts int, sleep time.Duration, f func() error) (err error) {
	for i := 0; i < attempts; i++ {
		if i > 0 {
			time.Sleep(sleep)
			sleep *= 2
		}

		err = f()
		if err == nil {
			return nil
		}
	}
	return fmt.Errorf("no luck after %d attempts: last error: %v", attempts, err)
}
```

```go
var bootstrapNode, node *P2PNode
retry(3, 0 *time.Seconds, func() error {
  bootstrapNode, node, err = setupTwoNodes(t, bootstrapNodeID, bootstrapNodeEndpoint, nodeID, nodeEndpoint, &PrivateKeyFromFile{bootstrapNodeSK}, &PrivateKeyFromFile{nodeSK})
  if err != nil {
    return err 
  }
})
```

Comment (adecaro) [reply platform/view/services/comm/p2p_test.go:124]: beautiful, let me do that

Comment (adecaro) [reply platform/view/services/comm/p2p_test.go]: done

File: platform/view/services/comm/p2p_test.go [hunk 2]
```diff
@@ -260,3 +274,18 @@ func SessionsForMPCTestRound(t *testing.T, bootstrapNode, node *P2PNode, bootstr
...
+			sleep *= 2
+		}
+
+		err = f()
+		if err == nil {
+			return nil
+		}
+	}
+	return fmt.Errorf("no luck after %d attempts: last error: %v", attempts, err)
+}
```
Comment (mbrandenburger) [inline platform/view/services/comm/p2p_test.go:291]: I suggest to put this function somewhere else, so we can easily reuse it.

Comment (adecaro) [reply platform/view/services/comm/p2p_test.go:291]: okay

Comment (adecaro) [reply platform/view/services/comm/p2p_test.go]: done :)

**Review by mbrandenburger — APPROVED** [review]
Thx Angelo. LGTM now

**Pull request merged**